support/bootstrap

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+#
+# Bootstraps the developer tools and development environment. This includes
+# copying project-standard git commit hooks that do things like ensure DCO
+# signoff is present during commit.
+
+# Best-effort check that we're in the envoy root directory.
```

Comment:
Can't you figure out the exact root using `git rev-parse --show-toplevel`?

Reply:
I'm not 100% sure that works in all the corner cases, but it looks like it works for submodules, which does seem to be a case we care about, so I'm ok doing this.